src/pages/Register/Register.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useCallback, ChangeEvent, FormEvent } from 'react';
+import { useSelector, useDispatch, useStore } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import BaseInput from 'components/BaseInput';
+import Button from 'components/Button';
+import Form from 'components/Form';
+import Layout from 'components/Layout';
+
+import { registerUser, Statuses } from 'store/slices/auth';
+import { State } from 'store';
+import { UserRegister } from 'types/user';
+
+const Register: React.FC = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const store = useStore();
+  const loading = useSelector<State>((state) => state.auth.loading);
+
+  const [fields, setFields] = useState<UserRegister>({
+    firstName: '',
+    secondName: '',
+    login: '',
+    email: '',
+    password: '',
+    phone: '',
+  });
+
+  const onSubmit = useCallback(
+    (e: FormEvent) => {
+      e.preventDefault();
+
+      if (!(fields.login && fields.password)) {
+        return;
+      }
+
+      dispatch(registerUser(fields));
+      if (store.getState().status === Statuses.OK) {
```

Comment:
❗️
Блоки `const`, `if` и `return` стоит отбивать пустой строкой
```suggestion
      dispatch(registerUser(fields));
      
      if (store.getState().status === Statuses.OK) {
```